src/common/src/array/struct_array.rs

```diff
@@ -219,6 +222,29 @@ impl Array for StructArray {
 }
 
 impl StructArray {
+    pub fn new(
+        bitmap: Bitmap,
+        children: Vec<ArrayRef>,
+        children_type: Arc<[DataType]>,
+        children_names: Arc<[String]>,
+        len: usize,
+    ) -> Self {
+        let heap_size = bitmap.estimated_heap_size()
```

Comment:
Underestimating `children_type`, `children_names`, `Vec` of `Arc`. Not sure how much this matters.

Reply:
`children_type`, `children_names` may be shared across different data chunks, so it's appropriate to count them. `Vec` of `Arc` is left for reason as above.

Reply:
Good point. This means `children` can also be shared because `ArrayRef` is `Arc`. Is this acceptable?

Reply:
> Good point. This means `children` can also be shared because `ArrayRef` is `Arc`. Is this acceptable?

Yes, it's possible, so the caller, e.g. the executor who will count its memory occupation should be responsible with this. It's hard without allocator.